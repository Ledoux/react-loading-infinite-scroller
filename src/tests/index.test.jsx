import React from 'react'
import { shallow } from 'enzyme'

import { RawLoadingInfiniteScroll } from '../index'

describe('RawLoadingInfiniteScroll', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = {
        query: {},
      }

      // when
      const wrapper = shallow(<RawLoadingInfiniteScroll {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
