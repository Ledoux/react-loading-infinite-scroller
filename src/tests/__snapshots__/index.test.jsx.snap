// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RawLoadingInfiniteScroll snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RawLoadingInfiniteScroll
    element="div"
    getScrollParent={null}
    hasMore={false}
    initialLoad={true}
    isLoading={false}
    isReverse={false}
    loader={null}
    pageStart={0}
    query={
      Object {
        "parse": [Function],
      }
    }
    ref={null}
    threshold={250}
    useCapture={false}
    useWindow={true}
  >
    <div />
  </RawLoadingInfiniteScroll>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
